src/gui/vsDevice.cpp

```diff
@@ -309,6 +347,15 @@ void VsDevice::terminateJackTrip()
 void VsDevice::onTextMessageReceived(const QString& message)
 {
     QJsonDocument newState = QJsonDocument::fromJson(message.toUtf8());
+
+    // We have a heartbeat from which we can read the studio auth token
+    // Use it to set up and start the pinger connection
+    QString token = newState["authToken"].toString();
```

Comment:
This seems to handle the case of going from empty authToken to non-empty authToken, but there's also the inverse we should tackle as well.

It's possible to go from non-empty authToken to empty authToken when you disconnect from a server. I think there's other code that probably handles stopping m_pinger when you disconnect but I think we'll want some guards around this section to make sure we don't restart it? At the very least, `m_pinger->start()` should fail if there's an empty token supplied

Reply:
added some additional validation on the `vsPinger` class. Now the caller cannot set/unset the token while the pinger is active, nor can it start the pinger before the token has been set.